src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/WorkspaceProjectContextHost.WorkspaceProjectContextHostInstance.cs

```diff
@@ -103,6 +108,16 @@ protected override async Task InitializeCoreAsync(CancellationToken cancellation
                             linkOptions: DataflowOption.PropagateCompletion,
                             cancellationToken: cancellationToken),
 
+                    ProjectDataSources.SyncLinkTo(
```

Comment:
This needs to be sync-linked with the ProjectBuildRuleSource, otherwise, you are going to be pushing changes to Roslyn that are out of sync in version. ie The command-line arguments will be out of sync with the other changes we're pushing to Roslyn, such as source files, references, etc.

Reply:
Done. This is implemented in the commit "Sync-link buildSnapshot with the ProjectBuildRuleSource"